tests/scale/test_load.py

```diff
@@ -156,36 +158,42 @@ def test_scaling_load(master_count,
                     range(min_index, max_index)]
    
     # create service accounts in parallel
+    sdk_security.install_enterprise_cli()
     service_account_threads = _spawn_threads(masters,
                                             _create_service_accounts,
                                             security=security_mode)
 
     thread_failures = _wait_and_get_failures(service_account_threads,
                                              timeout=SERVICE_ACCOUNT_TIMEOUT)
     # launch Jenkins services
-    install_threads = _spawn_threads(masters,
-                                     _install_jenkins,
-                                     event='deployments',
-                                     client=marathon_client,
-                                     external_volume=external_volume,
-                                     security=security_mode,
-                                     daemon=True)
-    thread_failures = _wait_and_get_failures(install_threads,
-                                             timeout=DEPLOY_TIMEOUT)
-    thread_names = [x.name for x in thread_failures]
-
-    # the rest of the commands require a running Jenkins instance
-    deployed_masters = [x for x in masters if x not in thread_names]
-    job_threads = _spawn_threads(deployed_masters,
-                                 _create_jobs,
-                                 jobs=job_count,
-                                 single=single_use,
-                                 delay=run_delay,
-                                 duration=work_duration,
-                                 scenario=scenario)
-    _wait_on_threads(job_threads, JOB_RUN_TIMEOUT)
-    r = json.dumps(TIMINGS)
-    print(r)
+    current = 0
+    end = max_index - min_index - 1
+    while current + batch_size <= end:
```

Comment:
Instead of making max_index mandatory, why not just use 
`for i <- current to current + batch_size` ? Also the default value of `max_index` seems to be `-1` and in that case end would be a negative value.

Reply:
this is to protect against the case where batch_size is not a factor of the size (max_index - min_index) for example I see a case where batch_size = 7 max_index=10, min_index = 0. so the masters list is inited with jenkins0 to jenkins10. iteration 1 deploys jenkins0 to jenkins6. if we do currrent + batch_size the next iteration will attempt jenkins7 to jenkins14 (going out of bounds of the list) the simple fix is to require max index as a required field, since the command will be entered once, it shouldnt be a problem for the operator to add this one extra flag. We could solve this problem in a cleaner way but taking the minimal fix.